### PR TITLE
Add M3 date range picker tests and fix header background theme color

### LINKED ISSUE #126826: Add tests for Material 3 Date pickers
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

While working on some other M3 date picker issues like https://github.com/flutter/flutter/issues/126597, I noticed the M3 layout and config changes made in https://github.com/flutter/flutter/pull/119033 aren't tested in `date_picker_test.dart`.

https://github.com/flutter/flutter/pull/119033/files#diff-198d5dd0fe89bb71e1940816baf20e7b57ee896ed31120b7c3fcebcc4ac3bfbe has a bunch of changes in `date_picker.dart` and most of these changes aren't tested in `date_picker_test.dart`.

Screenshot 2023-05-15 at 17 18 32

### Expected results

`date_picker_test.dart` has M3 test coverage

### Actual results

Most of the M3 changes aren't tested and modifying some M3 defaults or removing M3-related widgets from `date_picker.test` doesn't fail tests in the `date_picker_test.dart` class. 
Screenshot 2023-05-15 at 17 28 13

### Code sample

~

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

### LINKED ISSUE #127184: Date Range Picker doesn't use `DatePickerThemeData.rangePickerHeaderBackgroundColor` property
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

`showDateRangePicker` sample below contains a button to launch the date range picker and the `DatePickerThemeData` where color is provided to `rangePickerHeaderBackgroundColor` theme property.

### Expected results

The date range picker header background should use the provided theme color

Screenshot 2023-05-19 at 14 07 07

### Actual results

The date range picker header background doesn't use the provided theme color.

Screenshot 2023-05-19 at 14 07 17

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        useMaterial3: true,
        datePickerTheme: const DatePickerThemeData(
          rangePickerHeaderBackgroundColor: Colors.amber,
        ),
      ),
      home: Scaffold(
        body: Builder(builder: (BuildContext context) {
          return Center(
            child: ElevatedButton(
              onPressed: () async {
                await showDateRangePicker(
                  context: context,
                  initialDateRange: DateTimeRange(
                    start: DateTime(2021, DateTime.january, 15),
                    end: DateTime(2021, DateTime.january, 20),
                  ),
                  // initialDate: DateTime(2016, DateTime.january, 15),
                  firstDate: DateTime(2001),
                  lastDate: DateTime(2031, DateTime.december, 31),
                );
              },
              child: const Text('Show Date Range Picker'),
            ),
          );
        }),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

_No response_

### Logs

~

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[!] Flutter (Channel master, 3.11.0-6.0.pre.134, on macOS 13.3.1 22E772610a
    darwin-arm64, locale en-EE)
    • Flutter version 3.11.0-6.0.pre.134 on channel master at
      /Users/tahatesser/Code/flutter
    ! Upstream repository git@github.com:TahaTesser/flutter.git is not the same
      as FLUTTER_GIT_URL
    • FLUTTER_GIT_URL = git@github.com:NevercodeHQ/flutter.git
    • Framework revision f31dae2a80 (35 hours ago), 2023-05-17 17:18:51 -0700
    • Engine revision d970370779
    • Dart version 3.1.0 (build 3.1.0-118.0.dev)
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/tahatesser/Code/android-sdk
    • Platform android-33, build-tools 33.0.2
    • ANDROID_SDK_ROOT = /Users/tahatesser/Code/android-sdk
    • Java binary at: /Applications/Android Studio
      Preview.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.78.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] VS Code (version 1.79.0-insider)
    • VS Code at /Applications/Visual Studio Code - Insiders.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.3.1 22E772610a
      darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 113.0.5672.126

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

```

</details>

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/126826
fixes https://github.com/flutter/flutter/issues/127184

### Description
1. This adds some missing M3 Date Range Picker tests
2. Fix the header background color theme value is not, which was added during the M3 migration.

### Before

Screenshot 2023-05-19 at 14 07 17

### After 

Screenshot 2023-05-19 at 14 07 07

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/date_range_picker_test.dart [hunk 1]
```diff
@@ -1299,6 +1459,159 @@ void main() {
     await tester.tap(find.text('CANCEL'));
     await tester.pumpAndSettle();
   });
+
+
+  group('Material 2', () {
+    // Tests that are only relevant for Material 2. Once ThemeData.useMaterial3
```
Comment (HansMuller) [inline packages/flutter/test/material/date_range_picker_test.dart:1465]: We will not remove the tests until the support for M2 is removed. When M3 is the default, the support for M2 will continue. At some point after M3 becomes the default we'll deprecate the M2 support. Once another ~year has gone by (per the deprecation policy) we'll remove the M2 support.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/date_range_picker_test.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/test/material/date_range_picker_test.dart:1465]: This comment is from existing Material 2 group tests. I think we update all matches at once in the framework for consistency in a separate PR. WDYT?

Comment (HansMuller) [reply packages/flutter/test/material/date_range_picker_test.dart:1465]: SGTM

**Pull request merged**